mlir/test/Dialect/LLVMIR/global.mlir

```diff
@@ -117,11 +117,6 @@ llvm.mlir.global internal protected unnamed_addr @protected(42 : i32) : i32
 
 // -----
 
-// expected-error @+1 {{expects type to be a valid element type for an LLVM pointer}}
-llvm.mlir.global internal constant @constant(37.0) : !llvm.label
```

Comment:
Looks like we are regressing in the verifier here. Do you know whether there's a convention with `verify-diagnostics` tests to temporarily disable this line specifically and leave a todo?
I think something like `// todo-error` would work.

Reply:
Good catch. I'm moving the pointer type check logic directly into the `GlobalOp` verifier to restore this check.

Reply:
In that case you can probably use `TypeAttrOf` in `GlobalOp` instead of the plain `TypeAttr` to restrict the type that can be allocated

Reply:
The list is quite large, consisting of around 20 types.